holoviews/plotting/util.py

```diff
@@ -116,13 +132,18 @@ def compute_overlayable_zorders(obj, path=[]):
             # Skips branches of graph that collapse Overlay layers
             # to avoid adding layers that have been reduced or removed
             continue
+        elif depth is not None and depth < overlay_depth(inp):
+            # Skips branch of graph where the number of elements in an
+            # overlay has been reduced
+            continue
```

Comment:
Yes, it can't figure out which parts of the input operation should continue to be linked. A user may be able to supply that information but at that point its easier for them to declare their own Callable and list the inputs they need explicitly.